sql/core/src/test/resources/sql-tests/results/cte.sql.out

```diff
@@ -328,16 +328,891 @@ struct<scalarsubquery():int>
+
+
+-- !query 42
+WITH RECURSIVE r(level, data) AS (
+  VALUES (0, 'A')
+  UNION ALL
+  SELECT level + 1, r.data
+  FROM r
+  RIGHT JOIN (
+    SELECT 'B' AS data
+  ) AS o ON o.data = r.data
+)
+SELECT * FROM r
+-- !query 42 schema
+struct<>
+-- !query 42 output
+org.apache.spark.sql.AnalysisException
+Recursive reference r cannot be used here. This can be caused by using it on inner side of an outer join, using it with aggregate or distinct, using it in a subquery or using it multiple times in a recursive term (except for using it on different sides of an UNION ALL).;
+
+
+-- !query 43
+WITH RECURSIVE r(level, data) AS (
+  SELECT MAX(level) AS level, SUM(data) AS data FROM VALUES (0, 1), (0, 2)
+  UNION ALL
+  SELECT level + 1, data FROM r WHERE level < 10
+)
+SELECT * FROM r ORDER BY level
+-- !query 43 schema
+struct<>
+-- !query 43 output
+org.apache.spark.sql.AnalysisException
+cannot resolve '`level`' given input columns: [col1, col2]; line 2 pos 13
+
+
+-- !query 44
+WITH RECURSIVE r(group, data) AS (
+  VALUES (0, 1L)
+  UNION ALL
+  SELECT 1, SUM(data) FROM r WHERE data < 10 GROUP BY group
+)
+SELECT * FROM r
+-- !query 44 schema
+struct<>
+-- !query 44 output
+org.apache.spark.sql.AnalysisException
+Recursive reference r cannot be used here. This can be caused by using it on inner side of an outer join, using it with aggregate or distinct, using it in a subquery or using it multiple times in a recursive term (except for using it on different sides of an UNION ALL).;
+
+
+-- !query 45
+WITH RECURSIVE r(level) AS (
+  VALUES (1L)
+  UNION ALL
+  SELECT SUM(level) FROM r WHERE level < 10
+)
+SELECT * FROM r
+-- !query 45 schema
+struct<>
+-- !query 45 output
+org.apache.spark.sql.AnalysisException
+Recursive reference r cannot be used here. This can be caused by using it on inner side of an outer join, using it with aggregate or distinct, using it in a subquery or using it multiple times in a recursive term (except for using it on different sides of an UNION ALL).;
+
+
+-- !query 46
+WITH RECURSIVE r(level, data) AS (
+  VALUES (0, 'A')
+  UNION ALL
+  SELECT level + 1, data FROM r WHERE level < 10
+)
+SELECT COUNT(*) FROM r
+-- !query 46 schema
+struct<count(1):bigint>
+-- !query 46 output
+11
+
+
+-- !query 47
+WITH RECURSIVE r(level, data) AS (
+  VALUES (0, 'A')
+  UNION ALL
+  SELECT DISTINCT level + 1, data FROM r WHERE level < 10
+)
+SELECT * FROM r
+-- !query 47 schema
+struct<>
+-- !query 47 output
+org.apache.spark.sql.AnalysisException
+Recursive reference r cannot be used here. This can be caused by using it on inner side of an outer join, using it with aggregate or distinct, using it in a subquery or using it multiple times in a recursive term (except for using it on different sides of an UNION ALL).;
```

Comment:
```
 level | data 
-------+------
     0 | A
     1 | A
     2 | A
     3 | A
     4 | A
     5 | A
     6 | A
     7 | A
     8 | A
     9 | A
    10 | A
(11 rows)
```

Reply:
I will look into this one. If PostgreSQL allows it it should be allowed.

Reply:
Works as PostgreSQL from now.